velox/functions/lib/window/NthValue.cpp

```diff
@@ -38,14 +37,28 @@ class NthValueFunction : public exec::WindowFunction {
         isConstantOffsetNull_ = true;
         return;
       }
-      constantOffset_ =
-          args[1]
-              .constantValue->template as<ConstantVector<int64_t>>()
-              ->valueAt(0);
+      if (args[1].type->isInteger()) {
+        constantOffset_ =
+            args[1]
+                .constantValue->template as<ConstantVector<int32_t>>()
+                ->valueAt(0);
+      } else {
+        constantOffset_ =
+            args[1]
+                .constantValue->template as<ConstantVector<int64_t>>()
+                ->valueAt(0);
+      }
       VELOX_USER_CHECK_GE(
           constantOffset_.value(), 1, "Offset must be at least 1");
       return;
     }
+
+    if (args[1].type->isInteger()) {
```

Comment:
Move this to in the if (args[1].type->isInteger()) conditional statement above at line 40 ?